libbeat/template/load_integration_test.go

```diff
@@ -60,55 +61,227 @@ func defaultESLoader(t *testing.T) *ESLoader {
+	fields := testFields(t, name)
+
+	pathTestdata, err := filepath.Abs(filepath.Join("testdata"))
+	require.NoError(t, err)
+	fieldsYamlPath := filepath.Join(pathTestdata, "fields.yml")
+	fieldsJSONPath := filepath.Join(pathTestdata, "fields.json")
+
+	t.Run("load template failure", func(t *testing.T) {
+		for run, data := range map[string]struct {
+			cfg    TemplateConfig
+			info   beat.Info
+			fields []byte
+			err    string
+		}{
+			"loading disabled": {
+				cfg:  TemplateConfig{Enabled: false, Name: name},
+				info: beatInfo,
+			},
+			"invalid version": {
+				cfg:  TemplateConfig{Enabled: true, Name: name},
+				info: beat.Info{Version: "invalid", Beat: "testbeat", IndexPrefix: "testbeatidx"},
+				err:  "version",
+			},
+		} {
+			t.Run(run, func(t *testing.T) {
+				// Ensure template is not loaded
+				client.Request("DELETE", "/_template/"+name, "", nil, nil)
+				assert.False(t, loader.templateExists(name))
+
+				err := loader.Load(data.cfg, data.info, data.fields, false)
+				if data.err != "" {
+					if assert.Error(t, err) {
+						assert.Contains(t, err.Error(), data.err)
+					}
+				} else {
+					assert.NoError(t, err)
+				}
+
+				// Make sure template was not loaded
+				assert.False(t, loader.templateExists(name))
+			})
+		}
+	})
+
+	t.Run("template already exists", func(t *testing.T) {
+		cfg := TemplateConfig{Enabled: true, Name: name}
+
+		// Ensure template is not loaded
+		client.Request("DELETE", "/_template/"+name, "", nil, nil)
+		assert.False(t, loader.templateExists(name))
 
-	// Check for non existent template
-	assert.False(t, loader.templateExists("libbeat-notexists"))
+		// Load Template and check it exists
+		err := loader.Load(cfg, beatInfo, nil, false)
+		assert.NoError(t, err)
+		assert.True(t, loader.templateExists(name))
+		// check that source is enabled
+		tmpl := getTemplate(t, client, name)
+		assert.Equal(t, true, tmpl.SourceEnabled())
+
+		// Load template again, this time with custom settings
+		cfg.Settings = TemplateSettings{
+			Source: map[string]interface{}{
+				"enabled": false,
+			},
+		}
+		// Do not overwrite template as overwrite is not set to true
+		err = loader.Load(cfg, beatInfo, nil, false)
+		assert.NoError(t, err)
+		// check that template was not overwritten, by checking source is enabled
+		tmpl = getTemplate(t, client, name)
+		assert.Equal(t, true, tmpl.SourceEnabled())
+
+		// Do overwrite template
+		cfg.Overwrite = true
+		err = loader.Load(cfg, beatInfo, nil, false)
+		assert.NoError(t, err)
+		// check that template was not overwritten, by checking source is enabled
+		tmpl = getTemplate(t, client, name)
+		assert.Equal(t, false, tmpl.SourceEnabled())
+	})
+
+	t.Run("template exists for `name` but not for `json.name`", func(t *testing.T) {
+		// Load Template with name `name`
+		cfg := TemplateConfig{Enabled: true, Name: name}
+		err := loader.Load(cfg, beatInfo, nil, false)
+		assert.NoError(t, err)
+		assert.True(t, loader.templateExists(name))
```

Comment:
Looks like a common pattern, but:
- ensure we have a loader per test. The loader accepts a `*testing.T`. When running a test, do only use the `*testing.T` in the current context
- why doesn't loader.Load fail already? We need the error return somewhere? If so consider MustLoad and do not repeat the checks over and over again.